app/javascript/controllers/show_modal_controller.js

```diff
@@ -6,6 +6,6 @@ export default class extends Controller {
   copyFormTemplate(event) {
     const modalFormTemplate = event.target.nextElementSibling;
     this.containerTarget.innerHTML = modalFormTemplate.innerHTML;
-    this.titleTarget.innerText = event.target.innerText;
+    this.titleTarget.innerText = modalFormTemplate.dataset.title;
```

Comment:
Stop assuming the modal title will match the button that was clicked and take the more explicit approach of getting the title value out of the template. I suppose it could also be injected via `locals` when rendering the partial, but I kinda prefer having it live in the template. I couldn't defend that very strongly, so if someone has a reason to prefer injecting the value otherwise, I'm all :ear:.